contracts/test/v0.1/RequestResponseConsumerContract.test.ts

```diff
@@ -146,10 +146,446 @@ describe('Request-Response user contract', function () {
+    )
+
+    const fulfillReceipt = await (
+      await coordinatorContractOracleSigner.fulfillDataRequestBytes32(
+        requestId,
+        response,
+        requestCommitment,
+        isDirectPayment,
+        {
+          gasLimit: maxGasLimit + 300_000
+        }
+      )
+    ).wait()
+
+    expect(fulfillReceipt.events.length).to.be.equal(2)
+
+    // PREPAYMENT EVENT
+    const prepaymentEvent = prepaymentContract.interface.parseLog(fulfillReceipt.events[0])
+    expect(prepaymentEvent.name).to.be.equal('AccountBalanceDecreased')
+    expect(prepaymentEvent.args.accId).to.be.equal(accId)
+
+    // FIXME
+    // expect(prepaymentEvent.args.oldBalance).to.be.equal()
+    // expect(prepaymentEvent.args.newBalance).to.be.equal()
+
+    // COORDINATOR EVENT
+    const fulfillEvent = coordinatorContract.interface.parseLog(fulfillReceipt.events[1])
+    expect(fulfillEvent.name).to.be.equal('DataRequestFulfilledBytes32')
+    expect(fulfillEvent.args.requestId).to.be.equal(requestId)
+    expect(await consumerContract.s_responseBytes32()).to.be.equal(response)
+  })
+
+  it('Request & Fulfill Bytes', async function () {
+    const {
+      accId,
+      maxGasLimit,
+      consumerContract,
+      coordinatorContract,
+      prepaymentContract,
+      consumer,
+      rrOracle0
+    } = await loadFixture(deployFixture)
+    const prepaymentContractConsumerSigner = await ethers.getContractAt(
+      'Prepayment',
+      prepaymentContract.address,
+      consumer
+    )
+    const value = parseKlay(1)
+    await prepaymentContractConsumerSigner.deposit(accId, { value })
+    const requestReceipt = await (
+      await consumerContract.requestData(accId, maxGasLimit, {
+        gasLimit: 500_000
+      })
+    ).wait()
+
+    expect(requestReceipt.events.length).to.be.equal(1)
+    const requestEvent = coordinatorContract.interface.parseLog(requestReceipt.events[0])
+    expect(requestEvent.name).to.be.equal('DataRequested')
+
+    const eventArgs = [
+      'requestId',
+      'jobId',
+      'accId',
+      'callbackGasLimit',
+      'sender',
+      'isDirectPayment',
+      'data'
+    ]
+    for (const arg of eventArgs) {
+      expect(requestEvent.args[arg]).to.not.be.undefined
+    }
+
+    const { requestId } = requestEvent.args
+
+    const response = ethers.utils.formatBytes32String('hello')
+    const requestCommitment = {
+      blockNum: requestReceipt.blockNumber,
+      accId,
+      callbackGasLimit: maxGasLimit,
+      sender: consumerContract.address
+    }
+    const isDirectPayment = false
+
+    const coordinatorContractOracleSigner = await ethers.getContractAt(
+      'RequestResponseCoordinator',
+      coordinatorContract.address,
+      rrOracle0
+    )
+
+    const fulfillReceipt = await (
+      await coordinatorContractOracleSigner.fulfillDataRequestBytes(
+        requestId,
+        response,
+        requestCommitment,
+        isDirectPayment,
+        {
+          gasLimit: maxGasLimit + 300_000
```

Comment:
Why do we need to increase the `gasLimit`?

Reply:
Did I put it there? LOL!!! I think it's a random piece of code when I merged master branch. Anyway in tests, we can just keep it high to make sure the tx is executed.

Reply:
I think we should have a reason for having it here to mitigate confusion. Please try to remove if possible. If not possible we have to explain why this is needed.

Reply:
got it! now it gets weird. We can not run if we remove it, if we remove it and increase the gas limit, it can not run either. Can only run with `gasLimit: maxGasLimit + 300_000`
Let me find the route course

Reply:
Haha, that's funny 😃

Reply:
if we dont specify anything, it runs as well, maybe it's something with the `callwithexcatchgas()` function